maven-core/src/main/java/org/apache/maven/lifecycle/internal/MojoExecutor.java

```diff
@@ -220,20 +223,44 @@ private void execute( MavenSession session, MojoExecution mojoExecution, Project
      * TODO: ideally, the builder should take care of the ordering in a smarter way
      * TODO: and concurrency issues fixed with MNG-7157
      */
-    private static class ProjectLock implements AutoCloseable
+    private class ProjectLock implements AutoCloseable
     {
         final Lock acquiredAggregatorLock;
-        final Lock acquiredProjectLock;
+        final OwnerReentrantLock acquiredProjectLock;
 
-        ProjectLock( MavenSession session, MojoDescriptor mojoDescriptor, ReadWriteLock aggregatorLock )
+        ProjectLock( MavenSession session, MojoDescriptor mojoDescriptor )
         {
+            mojos.put( Thread.currentThread(), mojoDescriptor );
```

Comment:
this duplicates https://github.com/apache/maven/pull/736/files#diff-afa12cb60ed6ee8f6ad777fc0a0b4db1726b94306d7bda74f930ee95a27a9195R350. Intended?

Reply:
You're right.  I'll remove the access to the `mojos` map in the caller method and will add the corresponding removal from the map in the `close` method of the `ProjectLock`.  This way, only the `ProjectLock` class will actually access the map.